pkg/cli/tail_test.go

```diff
@@ -230,3 +391,19 @@ func (p *fakePuller) Pull(ctx context.Context, filter string, maxNum int) ([]*lo
 	p.gotMaxNum = maxNum
 	return p.logEntries, p.injectErr
 }
+
+func (p *fakePuller) StreamPull(ctx context.Context, filter string, logCh chan<- *loggingpb.LogEntry) error {
+	p.gotFilter = filter
+	if p.injectErr != nil {
+		return p.injectErr
+	}
+	for _, v := range p.logEntries {
+		logCh <- v
+	}
+
+	// mock real streamPull's behavior, looping until canceled
+	// instead of using a for{} that loops forever, this is
+	// more efficient in test
+	time.Sleep(3 * time.Second)
```

Comment:
The comment doesn't explain why the sleep is necessary. Can you elaborate?

Reply:
Comment updated.